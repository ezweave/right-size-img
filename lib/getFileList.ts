import { readdirSync } from 'fs';

export interface GetFileList {
	(path: string): Array<string> 
}

export const getFileList: GetFileList = (
	path
) => {
	const files = readdirSync(path);
	return files;
};