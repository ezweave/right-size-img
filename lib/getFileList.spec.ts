import { getFileList } from './getFileList';

import path from 'path';

const fixturePath = path.resolve(
	__dirname,
	'./__fixtures__'
);

describe('getFileList', () => {
	it('gets a list of files with a compatible extension', () => {
		expect(
			getFileList(fixturePath)
		).toMatchSnapshot();
	});
});