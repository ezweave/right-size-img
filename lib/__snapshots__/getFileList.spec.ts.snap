// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFileList gets a list of files with a compatible extension 1`] = `
Array [
  "large_rectangle.png",
  "large_square.png",
  "medium_square.png",
  "small_rectangle.png",
]
`;
